pkg/azuredisk/azuredisk.go

```diff
@@ -315,3 +322,34 @@ func (d *Driver) CreateManagedDisk(ctx context.Context, options *ManagedDiskOpti
 
 	return diskID, nil
 }
+
+// The name of the managed disk that is being created.
+// he name can't be changed after the disk is created.
+// Supported characters for the name are a-z, A-Z, 0-9 and _.
+// The maximum name length is 80 characters.
+//
+// See https://docs.microsoft.com/en-us/rest/api/compute/disks/createorupdate#uri-parameters
+func getValidDiskName(volumeName string) string {
+	diskName := strings.Replace(volumeName, "-", "_", -1)
+	if len(diskName) > diskNameMaxLength {
```

Comment:
add a warning log here since the name is truncated.

@andyzhangx any possible of same names after truncation?

Reply:
Done. 

In the next step, we will call azure cloud's api to create volume. In the api, the name will be checked if it exists.

Reply:
looks like the disk name could contain `-` , while according to doc, it said: 
```
 Supported characters for the name are a-z, A-Z, 0-9 and _. The maximum name length is 80 characters.
```
Let me ask azure disk team about this doc issue.

Reply:
@ZeroMagic Let's use the naming rule azure portal:
- _ . are all allowed chars